lms/djangoapps/courseware/tests/test_video_mongo.py

```diff
@@ -1079,64 +1079,81 @@ def test_source_is_empty(self):
     @ddt.data(
         (
             {
-                'desktop_webm': 'https://webm.com/dw.webm',
-                'hls': 'https://hls.com/hls.m3u8',
                 'youtube': 'v0TFmdO4ZP0',
-                'desktop_mp4': 'https://mp4.com/dm.mp4'
+                'hls': 'https://hls.com/hls.m3u8',
+                'desktop_mp4': 'https://mp4.com/dm.mp4',
+                'desktop_webm': 'https://webm.com/dw.webm',
             },
             ['https://www.youtube.com/watch?v=v0TFmdO4ZP0']
         ),
         (
             {
-                'desktop_webm': 'https://webm.com/dw.webm',
-                'hls': 'https://hls.com/hls.m3u8',
                 'youtube': None,
-                'desktop_mp4': 'https://mp4.com/dm.mp4'
+                'hls': 'https://hls.com/hls.m3u8',
+                'desktop_mp4': 'https://mp4.com/dm.mp4',
+                'desktop_webm': 'https://webm.com/dw.webm',
             },
-            ['https://hls.com/hls.m3u8']
+            ['https://www.youtube.com/watch?v=3_yD_cEKoCk']
```

Comment:
This behavior is not correct, if there is not youtube id from val OR external youtube url then HLS should get the priority.  Not the default url.

Reply:
Actually there is external YT id available in this case. When a video component is initialized this url is there as an external YT encoding.